tests/Hedgehog.Tests/TestDsl.fs

```diff
@@ -1,43 +1,38 @@
-module Hedgehog.Tests.TestDsl
+module internal Hedgehog.Tests.TestDsl
 
 #if FABLE_COMPILER
-
 open Fable.Mocha
 
-// Alias test functions so we do not have to deal with different open statements in every test file.
-let testCase = Fable.Mocha.Test.testCase
-let ptestCase = Fable.Mocha.Test.ptestCase
-let testList = Fable.Mocha.Test.testList
+let testCase = Test.testCase
+let ptestCase = Test.ptestCase
+let testList = Test.testList
 
 #else
-
 open Expecto
 
-// Alias test functions so we do not have to deal with different open statements in every test file.
-let testCase = Expecto.Tests.testCase
-let ptestCase = Expecto.Tests.ptestCase
-let testList = Expecto.Tests.testList
+let testCase = Tests.testCase
+let ptestCase = Tests.ptestCase
+let testList = Tests.testList
 
+type TestCase = Test
 #endif
 
-let inline testCases name testData testFun =
-    let nameWithData name data = sprintf "%s: (%A)" name data
-
-    [ for data in testData do
-        testCase (nameWithData name data) (fun _ ->
-            testFun data) ]
+let testCases (label : string) (xs : seq<'a>) (f : 'a -> unit) : List<TestCase> =
+    [ for x in xs do
+        testCase (sprintf "%s: (%A)" label x) (fun _ -> f x) ]
 
-/// Some tests are not running in javascript world.
-/// Use this to ignore such tests.
-let testCaseNoFable name testFun =
+let fableIgnore (label : string) (test : unit -> unit) : TestCase =
 #if FABLE_COMPILER
-    ptestCase name testFun
+    // Some tests are not running in Node.js.
+    ptestCase label test
 #else
-    testCase name testFun
+    testCase label test
 #endif
 
-let inline (=!) actual expected = Expect.equal expected actual "Should be equal"
+let inline (=!) (actual : 'a) (expected : 'a) : unit =
+    Expect.equal expected actual "Should be equal"
 
 [<RequireQualifiedAccess>]
 module Expect =
-    let inline isTrue value = Expect.isTrue value "Should be true"
+    let inline isTrue value =
```

Comment:
I'm not sure off the top of my head what's the IL diff between inlining vs no-inlining those. (I think, last time I had to deal with IL was [in 2012](https://github.com/AutoFixture/SemanticComparison/commit/af632fc95297b6bf1fa30a5cc0bf42472e46b773#diff-cb5da29d5d5e182dadee4df2aa3f982f4dc48c161e001feaf9045e6fd6baa79d) :open_mouth:, or [is it similar to GHC's optimizations](https://downloads.haskell.org/~ghc/8.6.3/docs/html/users_guide/glasgow_exts.html#inline-pragma)?)

Reply:
I'm completely in the dark there too, `inline` makes perfect sense to me when compiling, but how a compiled form retains the ability to be inlined is beyond me.

Reply:
Perhaps we can take those out. (Less is more(?))

Reply:
Since this test project isn't a library, I think inline here doesn't hurt anything. Usually `inline` is used to mitigate overhead, and to do fun ad-hoc things like we have in `Numeric.fs`. We don't really need either in this case, so it's up to you @moodmosaic.

Reply:
Let's remove it. I just forced-pushed on the branch.